### PR TITLE
Tabloid firebase auth

### PR DESCRIPTION
**Pull request by askingalot**
Here's a draft of the firebase auth chapter. It's...a start.

### CODE REVIEW

File: book-3-web-api/chapters/FIREBASE_AUTH.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+### Look around the repo
+
+In the repo you will find...
+
+* A Visual Studio solution and project both called `WisdomAndGrace`.
+* A `client` folder inside the project folder than contains a react application. This application's `package.json` file has already been configured to use the react proxy server.
+* A sql script for building the application database.
+* An `appsettings.json` file containing configuration information for the Web API.
+* A `.env` file containing configuration settings for the React application.
```
Comment (AdamSheaffer) [inline book-3-web-api/chapters/FIREBASE_AUTH.md:84]: Should we switch this to `.env.local`?

File: book-3-web-api/chapters/FIREBASE_AUTH.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+### Update the Repo
+
+Before we can run the app we need to make a few changes.
+
+1. Open the SQL script in an editor and find the `INSERT` statements that insert records into the `UserProfile` table. Modify these statements to add the users you added to Firebase earlier in this chapter.
+1. Update the `appsettings.json` file. Change the value of the `FirebaseProjectId` key to me the Project Id of your Firebase project.
+1. Update the `.env` file. Change the value of the `REACT_APP_API_KEY` key to the API Key from your Firebase project.
```
Comment (AdamSheaffer) [inline book-3-web-api/chapters/FIREBASE_AUTH.md:92]: `.env.local`?

File: book-3-web-api/chapters/FIREBASE_AUTH.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+#### In the Gifter Database
+
+> **NOTE:** The following will delete and recreate the database.
+
+1. Update the Gifter SQL script to add a `FirebaseUserId` column to the `UserProfile` table.
+
+    ```sql
+    CREATE TABLE [UserProfile] (
+    [Id] INTEGER PRIMARY KEY IDENTITY NOT NULL,
+    [FirebaseUserId] NVARCHAR(28) NOT NULL,
```
Comment (AdamSheaffer) [inline book-3-web-api/chapters/FIREBASE_AUTH.md:134]: I don't think it matters much, but can we add a `UNIQUE` here so we get sqlserver to add an index?

**Comment by AdamSheaffer** [comment]
In the `UserProvider` in WisdomAndGrace can we switch this line to use a ternary instead?

```js
{(isFirebaseReady && props.children) || <Spinner className="app-spinner dark"/> }
```

@MoSilvera , they probably don't use short circuiting like this very often do they?

**Comment by askingalot** [comment]
I added the unique constraint. Thanks for catching that.

I changed the spinner code to use a ternary. I was just having some fun last night when I wrote it that way, but a ternary is clearly the better approach.

I'm torn on the `.env.local` thing I'm inclined to save that for the group project boilerplate. At that time we'll also need to deal with local `appsettings.json`. I think we can setup the boilerplate with appropriate .gitignore files and "sample" versions of the `.env.local` and maybe `appsettings.local.json`.

**Pull request merged**